Remove Quirk shouldIgnoreContentObservationForSyntheticClick for shutterstock.com

karlcow opened the pull request:
#### 3c8d3201425b822a3934cc3474af0be1bd4fb490
<pre>
Remove Quirk shouldIgnoreContentObservationForSyntheticClick for shutterstock.com
<a href="https://bugs.webkit.org/show_bug.cgi?id=260006">https://bugs.webkit.org/show_bug.cgi?id=260006</a>
rdar://107623325

Reviewed by Brent Fulgham.

Shutterstock in landscape mode needed two clicks to be able to enter
a search input in the search bar when in landscape mode. The quirk was
fixing this issue.

This patch removes the quirk which is not necessary anymore as we can
search at the first tap after deactivating Site Specific Hacks.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldIgnoreContentObservationForSyntheticClick const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::handleSyntheticClick):

Canonical link: <a href="https://commits.webkit.org/266776@main">https://commits.webkit.org/266776@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/092b7ffdd1f4382b914cb63ced4a5cab54837b74

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14692 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15002 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15351 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16440 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13858 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/14830 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17520 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15085 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/16501 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14873 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15377 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12475 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17175 "Built successfully") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12660 "2 flakes 7 failures") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13253 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/20249 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13735 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13420 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16659 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13965 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11810 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13266 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3553 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17601 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13811 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

karlcow commented on Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
```diff
@@ -769,7 +769,7 @@ static void dispatchSyntheticMouseMove(LocalFrame& mainFrame, const WebCore::Flo
     auto isFirstSyntheticClickOnPage = !m_hasHandledSyntheticClick;
```
I need to remove this instance too, as its only usage is in the context of this quirk.

karlcow replied:
done.

brentfulgham reviewed (approved):
It's wonderful seeing so many Quirks being removed. r=me!

karlcow committed:
Remove Quirk shouldIgnoreContentObservationForSyntheticClick for shutterstock.com
https://bugs.webkit.org/show_bug.cgi?id=260006
rdar://107623325

Reviewed by Brent Fulgham.

Shutterstock in landscape mode needed two clicks to be able to enter
a search input in the search bar when in landscape mode. The quirk was
fixing this issue.

This patch removes the quirk which is not necessary anymore as we can
search at the first tap after deactivating Site Specific Hacks.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldIgnoreContentObservationForSyntheticClick const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::handleSyntheticClick):

Canonical link: https://commits.webkit.org/266776@main
